Sources/SKTestSupport/IndexedSingleSwiftFileWorkspace.swift

```diff
@@ -0,0 +1,125 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import SKCore
+import Foundation
+import ISDBTibs
+import SourceKitLSP
+import TSCBasic
+
+public struct IndexedSingleSwiftFileWorkspace {
+  enum Error: Swift.Error {
+    case swiftcNotFound
+  }
+
+  public let testClient: TestSourceKitLSPClient
+  public let fileURI: DocumentURI
+  public let positions: DocumentPositions
+
+  public init(
+    _ markedText: String,
+    testName: String = #function
+  ) async throws{
```

Comment:
Incidentally, that’s on of the motivation for this PR. So we can format the entire repository without the need for a `format.py` equivalent that determines which files should be formatted. Currently we can’t do that because we don’t want to format the `INPUTS` folder.